project/scripts/fetch_trends.py

```diff
@@ -18,26 +18,31 @@
 
 import pandas as pd
 from pytrends.request import TrendReq
+
 pytrends = TrendReq(tz=0) #tz=0 puts us on UTC
 
-from dates import get_start_times, get_end_times, date_to_epoch
+from dates import get_start_times, get_end_times, date_to_epoch, get_all_dates
+from data_generator import get_missing_data_point
 
 NUM_TRENDING = 10 # number of trending searches to return
 
 
-def get_updated_daily_data(search_term, investment_date):
+def get_updated_daily_data(entity):
```

Comment:
Nice, why stop short of a class, speed?. Not saying to change it

Reply:
A class representing what exactly? The entity object is built in.

Reply:
entity seems like a dict things are keyed by string, and potential to go wrong vs a class whether things are accessed by fields

Reply:
If I convert to a class one of the attributes will need to be a dictionary of date - data pairs anyway, since the dates themselves aren't going to be common to all objects and change each day. I'm not sure what I'd gain from this, but for what it's worth this is how it looks in the datastore for Python documentation.